File: backend/internal/domain/location.go
package domain

import "time"

type Point struct {
	Lat float64
	Lon float64
}

type Location struct {
	ResourceMeta
	TripID           string
	Name             string
	City             string
	Country          string
	Coordinates      Point
	Notes            string
	Sequence         int
	ShortDescription string
	DateFrom         time.Time
	DateTo           time.Time
}

File: frontend/types/location.ts
import { components } from "@/generated/types";

export type LocationResponse = components["schemas"]["LocationResponse"];
export type CreateLocationRequest = components["schemas"]["CreateLocationRequest"];
export type UpdateLocationRequest = components["schemas"]["UpdateLocationRequest"];

export type Location = {
    name: string;
    city: string;
    country: string;
    countryCode: string;
    shortDescription: string;
    dateFrom: string;
    dateTo: string;
    latitude?: number;
    longitude?: number;
    notes?: string;
}